src/main/java/edu/harvard/iq/dataverse/api/DatasetFieldServiceApi.java

```diff
@@ -543,4 +536,86 @@ public static String getDataverseLangDirectory() {
         return dataverseLangDirectory;
     }
 
+    @POST
+    @Consumes("text/tab-separated-values")
+    @Path("mergeLanguageList")
+    public Response mergeLanguagePropertyFile(File file) {
+        ActionLogRecord alr = new ActionLogRecord(ActionLogRecord.ActionType.Admin, "mergeLanguageList");
+        alr.setInfo( file.getName() );
+        JsonArrayBuilder responseArr = Json.createArrayBuilder();
+
+        BufferedReader br = null;
+        String line = null;
+        String splitBy = "\t";
+        int lineNumber = 0;
+        int offset = 200; // number of existing languages // TODO: get the number from db
+
+        DatasetFieldType dsv = datasetFieldService.findByName("language");
+
+        try {
+            // Load the new language tab file
+            // The format of the file is: Id<tab>Part2b<tab>Part2t<tab>Part1<tab>Scope<tab>Language_Type<tab>Ref_Name<tab>Comment
+            // Example: aaa\t\t\t\tI\tL\tGhotuo
+            logger.fine("Loading ISOTabFile from " + file);
+
+            br = new BufferedReader(new FileReader("/" + file));
+            br.readLine(); // ignore first line as it is a header
+            while ((line = br.readLine()) != null) {
+                String[] fields = line.split(splitBy);
+                if (fields.length >= 7) {
+                    String code = fields[0];
+                    Set<String> codes = new HashSet<>();
+                    codes.addAll(List.of(code, fields[1], fields[2], fields[3]));
+                    codes.removeAll(Arrays.asList("", null));
+                    String name = fields[6];
+                    // get controlledvocabularyvalue by name or by codes
+                    ControlledVocabularyValue cvv = datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(dsv, name, true);
+                    // if not found, find using main code and alternate codes
+                    Iterator<String> codesIterator = codes.iterator();
+                    while (cvv == null && codesIterator.hasNext()) {
+                        cvv = datasetFieldService.findControlledVocabularyValueByDatasetFieldTypeAndStrValue(dsv, codesIterator.next(), true);
+                    }
+
+                    // if it is found we need to merge the alternate codes since the next step will delete all existing alternate codes before adding the new ones
```

Comment:
The merge does not remove the original codes since they may not be in the new file. This is why the original alts get saved and merged to the new list

Reply:
OK - so this is a problem for Pular for example.